codegen/masm/src/codegen.rs

```diff
@@ -136,111 +153,168 @@ impl<'ast> CodeGenerator<'ast> {
 
     /// Emits code for the procedure `cache_z_exp`.
     ///
-    /// The emitted procedure computes a `z**exp` for each periodic column, these values are later
-    /// on used to evaluate the polynomial of each periodic column.
+    /// The procedure computes and caches the necessary exponentiation of `z`. These values are
+    /// later on used to evaluate each periodic column polynomial and the constraint divisor.
     ///
-    /// The emitted code is optimized to performn the fewest number of exponentiations, this is
-    /// achieved by observing that periodic columns and trace length are both power-of-two in
-    /// length, since the exponent is defined as `exp = trace_len / periodic_column_len`, all
-    /// exponents are themselves powers-of-two.
+    /// This procedure exists because the VM doesn't have native instructions for exponentiation of
+    /// quadratic extension elements, and this is an expensive operation.
     ///
-    /// The algorithm computes the exponents from smallest to largest, using the previous result as
-    /// a cache value.
+    /// The generated code is optimized to perform the fewest number of exponentiations, this is
+    /// achieved by observing that periodic columns and trace length are both powers-of-two, since
+    /// the exponent is defined as `exp = trace_len / periodic_column_len`, all exponents are
+    /// themselves powers-of-two. This allows the results to be computed from smallest to largest,
+    /// re-using the intermediary values.
     fn gen_cache_z_exp(&mut self) -> Result<(), CodegenError> {
```

Comment:
Should we move this function into a separate file? It could still be a part of the `CodeGenerator` impl - defined in its own file. It actually may make sense to create a file for all "boilerplate" code generation (e.g., `boilerplate.rs`).

Reply:
This was requested on a previous PR https://github.com/0xPolygonMiden/air-script/pull/316#discussion_r1234892268 , I have done that, but in a commit downstream. I would prefer to not do any code reorganization now since that is not the goal of this PR and that will save me work on conflicts later on.

Reply:
Yep - sounds good!